src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -534,6 +534,30 @@ private boolean hasEmptyLineBefore(DetailAST token) {
     }
 
     /**
+     * Check if token is preceded by javadoc comment
+     * @param token token for check.
+     * @return true, if token is predicted by javadoc comment.
+     */
+    private boolean isPrecededByJavadoc(DetailAST token) {
+        final int lineNo = token.getLineNo();
+        boolean result = true;
+        //  [lineNo - 2] is the number of the previous line because the numbering starts
+        //  from zero.
+        int lineIndex = lineNo - 2;
+        final String[] lines = getLines();
+        String lineBefore = lines[lineIndex].trim();
+        while (!lineBefore.startsWith("/**")) {
+            if (lineIndex == 0 || !lineBefore.isEmpty() && lineBefore.charAt(0) != '*') {
```

Comment:
yes, we have to support both approaches to write javadoc.

current implementations is fragile by design, as this Check is not comments aware - https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

to receive all comments well parsed Check need to override method 
```
    @Override
    public boolean isCommentNodesRequired() {
        return true;
}
```
Example:
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/TodoCommentCheck.java#L68

in this case Check will have all comments in good form.
Useful method - https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/utils/JavadocUtils.java#L246

I will write a bit more about isCommentNodesRequired in scope of #3561